src/plugins/github/graphql.js

```diff
@@ -46,9 +119,235 @@ export function createQuery(): Body {
+          b.field(
+            "pullRequests",
+            {
+              first: b.literal(PAGE_LIMIT),
+              after: b.literal(result.pullRequests.pageInfo.endCursor),
+            },
+            [b.fragmentSpread("prs")]
+          ),
+        ]),
+      ],
+      destinationPath: path,
+    };
+  }
+  if (result.issues) {
+    for (let i = 0; i < result.issues.nodes.length; i++) {
+      const issue = result.issues.nodes[i];
+      const subpath = [...path, "issues", "nodes", i];
+      yield* continuationsFromIssue(issue, issue.id, subpath);
+    }
+  }
+  if (result.pullRequests) {
+    for (let i = 0; i < result.pullRequests.nodes.length; i++) {
+      const pr = result.pullRequests.nodes[i];
+      const subpath = [...path, "pullRequests", "nodes", i];
+      yield* continuationsFromPR(pr, pr.id, subpath);
+    }
+  }
+}
+
+function* continuationsFromIssue(
+  result: any,
+  nodeId: string,
+  path: $ReadOnlyArray<string | number>
+): Iterator<Continuation> {
+  const b = build;
+  if (result.comments.pageInfo.hasNextPage) {
+    yield {
+      enclosingNodeType: "ISSUE",
+      enclosingNodeId: nodeId,
+      selections: [
+        b.inlineFragment("Issue", [
+          b.field(
+            "comments",
+            {
+              first: b.literal(PAGE_LIMIT),
+              after: b.literal(result.comments.pageInfo.endCursor),
+            },
+            [b.fragmentSpread("comments")]
+          ),
+        ]),
+      ],
+      destinationPath: path,
+    };
+  }
+}
+
+function* continuationsFromPR(
+  result: any,
+  nodeId: string,
+  path: $ReadOnlyArray<string | number>
+): Iterator<Continuation> {
+  const b = build;
+  if (result.comments && result.comments.pageInfo.hasNextPage) {
+    yield {
+      enclosingNodeType: "PULL_REQUEST",
+      enclosingNodeId: nodeId,
+      selections: [
+        b.inlineFragment("PullRequest", [
+          b.field(
+            "comments",
+            {
+              first: b.literal(PAGE_LIMIT),
+              after: b.literal(result.comments.pageInfo.endCursor),
+            },
+            [b.fragmentSpread("comments")]
+          ),
+        ]),
+      ],
+      destinationPath: path,
+    };
+  }
+  if (result.reviews && result.reviews.pageInfo.hasNextPage) {
+    yield {
+      enclosingNodeType: "PULL_REQUEST",
+      enclosingNodeId: nodeId,
+      selections: [
+        b.inlineFragment("PullRequest", [
+          b.field(
+            "reviews",
+            {
+              first: b.literal(PAGE_LIMIT),
+              after: b.literal(result.reviews.pageInfo.endCursor),
+            },
+            [b.fragmentSpread("reviews")]
+          ),
+        ]),
+      ],
+      destinationPath: path,
+    };
+  }
+  if (result.reviews) {
+    for (let i = 0; i < result.reviews.nodes.length; i++) {
+      const issue = result.reviews.nodes[i];
+      const subpath = [...path, "reviews", "nodes", i];
+      yield* continuationsFromReview(issue, issue.id, subpath);
```

Comment:
it's nice that we can use nested iterators this way.

Reply:
Yep! As far as I can tell, the iterator and generator APIs are pretty well designed.